modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/ShareFragment.kt

```diff
@@ -4,124 +4,264 @@ import android.os.Bundle
-                SharePodcastHelper(
-                    podcast,
-                    episode,
-                    null,
-                    requireContext(),
-                    ShareType.EPISODE,
-                    SourceView.PLAYER,
-                    analyticsTracker
-                ).showShareDialogDirect()
+            viewModel.playingEpisodeLive.observe(viewLifecycleOwner) { (_, backgroundColor) ->
+                applyColor(theme, backgroundColor)
             }
-            close()
         }
-        binding.buttonShareCurrentPosition.setOnClickListener {
-            if (podcast != null && episode is PodcastEpisode) {
-                SharePodcastHelper(
-                    podcast,
-                    episode,
-                    episode.playedUpTo,
-                    requireContext(),
-                    ShareType.CURRENT_TIME,
-                    SourceView.PLAYER,
-                    analyticsTracker
-                ).showShareDialogDirect()
+        (dialog as? BottomSheetDialog)?.behavior?.state = BottomSheetBehavior.STATE_EXPANDED
+    }
+}
+
+@Composable
+private fun ShareScreen(
+    podcast: Podcast?,
+    episode: BaseEpisode?,
+    backgroundColor: Color,
+    onSharePodcast: () -> Unit,
+    onShareEpisode: () -> Unit,
+    onShareCurrentPosition: () -> Unit,
+    onShareOpenFileIn: () -> Unit,
+) {
+    Surface(
+        modifier = Modifier
+            .fillMaxWidth(),
+        color = backgroundColor
+    ) {
+        Column(
+            modifier = Modifier
+                .padding(bottom = 24.dp)
+        ) {
+            Pill(
+                modifier = Modifier
+                    .align(Alignment.CenterHorizontally)
+            )
+            if (podcast != null) {
+                ShareButton(
+                    modifier = Modifier
+                        .padding(top = 20.dp),
+                    label = IR.string.share_podcast,
+                    onClick = onSharePodcast
+                )
             }
-            close()
-        }
-        binding.buttonOpenFileIn.setOnClickListener {
-            if (podcast != null && episode is PodcastEpisode) {
-                SharePodcastHelper(
-                    podcast,
-                    episode,
-                    episode.playedUpTo,
-                    requireContext(),
-                    ShareType.EPISODE_FILE,
-                    SourceView.PLAYER,
-                    analyticsTracker
-                ).sendFile()
+            Divider(
+                color = Color.LightGray.copy(0.3f),
+                thickness = 1.dp,
+            )
+            Divider(thickness = 1.dp)
+            if (podcast != null && episode is BaseEpisode) {
+                ShareButton(
+                    label = IR.string.podcast_share_episode,
+                    onClick = onShareEpisode
+                )
+                Divider(
+                    color = Color.LightGray.copy(0.3f),
```

Comment:
We should consider to use `MaterialTheme` outline color as divider colours rather than hardcoding them. Wdyt?